### ISSUE TITLE
A specific wording in `MAKE-ARRAY`.

### ISSUE THREAD

**Issue by jcguu95** [s-expressionists/wscl#53]
[CLHS: Function MAKE-ARRAY](https://www.lispworks.com/documentation/HyperSpec/Body/f_mk_ar.htm) says

> If initial-element is not supplied, the consequences of later reading an **uninitialized element** of new-array are undefined unless either initial-contents is supplied or displaced-to is non-nil. 

An element in the spec just means a lisp datum. It does not make sense to talk about whether an element is initialized or not. It is better, in my opinion, to instead say

> If initial-element is not supplied, the consequences of later reading an **uninitialized slot** of new-array are undefined unless either initial-contents is supplied or displaced-to is non-nil. 

But then we need to add a definition of a slot for an array somewhere else too.

**Comment by Bike** [comment]
Yeah, I wouldn't want to introduce new terminology, or rather to reuse "slot" to mean something other than a field in a structure or standard object. I'm not sure what wording would be better, though. "element" is already defined as "an [object](http://clhs.lisp.se/Body/26_glo_o.htm#object) that is an [element](http://clhs.lisp.se/Body/26_glo_e.htm#element) of the [list](http://clhs.lisp.se/Body/26_glo_l.htm#list) or [array](http://clhs.lisp.se/Body/26_glo_a.htm#array) that is the [sequence](http://clhs.lisp.se/Body/26_glo_s.htm#sequence)" which is uh, a little recursive but still seems more appropriate than "slot".

The issue here is that it's more the position in the array that's "uninitialized", rather than there being some kind of distinguished uninitialized object. Maybe just dropping "uninitialized" would be good...?

**Comment by jcguu95** [comment]
Dropping "uninitialized" is a bit restrictive too, because the user may write elements to such an array before reading it. In that case, reading the place afterward should not be undefined.

Actually, "slot" may not be a bad choice because [its definition in the glossary](https://www.lispworks.com/documentation/HyperSpec/Body/26_glo_s.htm) doesn't restrict it to be used only in the context of standard objects:

```
slot n. a component of an object that can store a value. 
```

**Comment by robert-strandh** [comment]
Given the glossary definition, I think "slot" is good.  I use it myself to talk about the CAR and the CDR
of a CONS cell.

**Comment by jcguu95** [comment]
Sure, hence I think the suggested edition in OP works.

> If initial-element is not supplied, the consequences of later reading an uninitialized slot of new-array are undefined unless either initial-contents is supplied or displaced-to is non-nil.
